libraries/GCS_MAVLink/GCS_Common.cpp

```diff
@@ -173,6 +173,8 @@ GCS_MAVLINK::setup_uart(const AP_SerialManager& serial_manager, AP_SerialManager
         // after experiments with MAVLink2
         status->flags |= MAVLINK_STATUS_FLAG_OUT_MAVLINK1;
     }
+
+    set_mavlink_message_id_interval(MAVLINK_MSG_ID_HEARTBEAT, 1000);
```

Comment:
This shouldn't be here; under the call to `initialise_message_intervals_from_streamrates()` would be more appropriate.

Reply:
fixed